File: mjcbsn22/Equal_Sides_Of_An_Array.py
def find_even_index(arr):
    for index, int in enumerate(arr):
        left = sum_range(arr, 0, index)
        right = sum_range(arr, index, len(arr))
        if left == right:
            return index
    return -1


def sum_range(arr, a, b):
    return sum(arr[a:b + 1])

File: kminehart/equal_sides_of_an_array.c
int find_even_index(const int *values, int length) {
  int left = 0;
  int right = 0;
  
  for(int i = 0; i < length; i++) {
    if(i == 0) {
      left = 0;
    } else {
      left += values[i - 1];
    }
    right = 0;
    
    for(int x = i + 1; x < length; x++) {
      right += values[x];
    }
    
    if(left - right == 0) {
      return i;
    }
  }
  
  return -1;
}
